app/src/main/java/com/example/evently/ui/entrant/SearchEventsListFragment.java

```diff
@@ -0,0 +1,76 @@
+package com.example.evently.ui.entrant;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+import android.util.Log;
+import android.widget.Toast;
+import androidx.navigation.NavController;
+import androidx.navigation.fragment.NavHostFragment;
+
+import com.google.firebase.Timestamp;
+
+import com.example.evently.data.EventsDB;
+import com.example.evently.data.model.Event;
+import com.example.evently.ui.common.EventsFragment;
+
+/**
+ * A fragment representing list of events filtered by the search bar
+ */
+public class SearchEventsListFragment extends EventsFragment {
+    private List<Event> eventList = new ArrayList<>();
+    public List<Event> filteredList = new ArrayList<>();
+
+    public void filter(String text) {
+        if (adapter == null || text.isEmpty()) {
+            adapter.updateEvents(eventList);
```

Comment:
I'm guessing this is a mistake since this would raise a `NullPointerException` (`adapter` may be `null`).

Surprised android studio doesn't give a lint for this.